### DISCUSSION TITLE
Having issues with importing into project

### DISCUSSION THREAD

**Discussion by bjazmoore** [hajimehoshi/ebiten#1512] [Q&A] [answered]
I am pretty new to Go, so I might just be doing something stupid.  

I have installed Ebiten with the command: go get github.com/hajimehoshi/ebiten/v2 - Windows environment.  I notice it has installed into pkg\mod\github.com\hajimehoshi\ebiten\v2@2.0.6.  I am using Visual Code with linter installed.   The linter keeps complaining that it cannot find the package (See image).
![shot_210224_234725](https://user-images.githubusercontent.com/43559858/109109553-1ec49280-76fb-11eb-87db-203f35f3f8b4.png)

It is a pretty new installation and I might have my paths messed up, but I don't think so.

**Comment by AWachtendorf** [comment, marked as answer]
Hi!

Starting out with go can be somewhat confusing. Did you create a go.mod,and checked if gomodules are enabled? (check "go env" in terminal?)

**Reply by bjazmoore** [reply]
I guess I do not understand the go.mod functionality.  I did just now add the mod and issued the go mod tidy command and it all looks good.  In all the books and videos I have been watching no one talks about go.mod.  Where can I learn about it (in plain english)?  Thanks

**Reply by hajimehoshi** [reply]
I'm not familiar with Visual Studio Code, but
* Which version of Go are you using?
* Are there `go.mod` and `go.sum` in your workspace?

For the first Go module usage, https://golang.org/doc/tutorial/getting-started should be a good document.

**Reply by AWachtendorf** [reply]
"Go used to rely on the Gopath as a directory for all go projects. This proved to be not the best solution so go invented the go.mod to work with projects outside of gopath. The go.mod manages your dependencies. For example when ebiten was updated from 2.0.0.5 to 2.0.0.6 I just changed that in my go mod and it synced automatically.
I really dislike that about many tutorials, the Gopath is nearly redundant since some time now, still it's mentioned in a lot of beginner guides. "

**Reply by bjazmoore** [reply]
Thanks everyone for your help.  Working on getting modules down.  This issue is closed.  It is working as it should.
